src/index.ts

```diff
@@ -150,6 +151,145 @@ app.get("/api/users/:id", async (req, res) => {
   }
 });
 
+
+/**
+ * @openapi
+ * /api/users: 
+ *   delete:
+ *     summary: Removes user
+ *     responses:
+ *       200:
+ *         description: Successfully removed a user
+ *         content:
+ *           text/plain:
+ *             schema:
+ *               type: string
+ *       400:
+ *         description: Id has wrong data type
+ *         content:
+ *           text/plain:
+ *             schema:
+ *               type: string
+ *       404:
+ *         description: No user with given id
+ *         content:
+ *           text/plain:
+ *             schema:
+ *               type: string
+ *   parameters:
+ *   - name: id
+ *     in: path
+ *     description: Id of the user to be removed 
+ *     required: true
+*/
+
+app.delete("/api/users/:id", async (req, res) => {
+  const { id } = req.params;
+
+  if (typeof id !== "string") {
+    res
+      .status(400)
+      .send("Id has wrong data type");
+  }
+
+  try {
+    const user = await prisma.user.delete({
+      where: {
+        id: Number(id),
+      },
+    });
+
+    res.send(`User with id ${id} was deleted`);
+  } catch (e) {
+    res.status(404).send(`There is no user with id ${id}`);
+
+  }
+});
+
+/**
+ * @openapi
+ * /api/users: 
+ *   put:
+ *     summary: Updates user data 
+ * 
+ *     requestBody:
+ *       content:
+ *         application/json:
+ *           schema:
+ *             type: object
+ *             name: user data
+ *             required: true
+ *             properties:
+ *               name:
+ *                 type: string
+ *                 description: User name
+ *                 example: John Doe
+ *               email:
+ *                 type: string
+ *                 description: User email
+ *                 example: test@test.pl
+ *
+ *     responses:
+ *       200:
+ *         description: Successfully removed a user
+ *         content:
+ *           text/plain:
+ *             schema:
+ *               type: string
+ *       400:
+ *         description: Data are not provided or Id has wrong data type
+ *         content:
+ *           text/plain:
+ *             schema:
+ *               type: string
+ *       404:
+ *         description: No user with given id
+ *         content:
+ *           text/plain:
+ *             schema:
+ *               type: string
+ *     parameters:
+ *     - name: id
+ *       in: query
+ *       description: Id of the user to be updated 
+ *       required: true
+*/
+
+app.put("/api/users/:id", async (req, res) => {
+  const { id } = req.params;
+  const { data } = req.body;
+
+  if (typeof id !== "string") {
+    res
+      .status(400)
+      .send("Id has wrong data type. It should be string.");
+  }
+
+  if (!data) {
+    res
+      .status(400)
+      .send("Data are not provided");
+  }
+
+  const { name, email } = data;
+
+  try {
+    const user = await prisma.user.update({
+      where: {
+        id: Number(id)
+      },
+      data: {
+        name,
+        email,
+      }
```

Comment:
If it's a PUT endpoint that we're talking about here - it should always **replace whole entity when updating it**. In opposition to PATCH, which can do a partial update. Have a look here: https://www.geeksforgeeks.org/difference-between-put-and-patch-request/

So, to at least get the functionality of replacing whole entity you could've done something like this:
```suggestion
      data: {
        name: name ?? null,
        email: email ?? null,
      }
```

Read [this guide](https://www.prisma.io/docs/concepts/components/prisma-client/null-and-undefined) to understand the difference between `undefined` and `null` in prisma

Reply:
Thx for the guidance.